tests/units/Impact.class.php

```diff
@@ -0,0 +1,287 @@
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+*/
+
+namespace tests\units;
+
+class Impact extends \GLPITestCase {
+
+   private function clearImpactItems() {
+      global $DB;
+
+      $DB->delete(\ImpactItem::getTable(), [true]);
+      $DB->delete(\ImpactRelation::getTable(), [true]);
+      $DB->delete(\ImpactCompound::getTable(), [true]);
+   }
+
+   public function testGetTabNameForItem_notCommonDBTM() {
+      $this->clearImpactItems();
+      $impact = new \Impact();
+
+      $this->exception(function() use ($impact) {
+         $notCommonDBTM = new \Impact();
+         $impact->getTabNameForItem($notCommonDBTM);
+      })->isInstanceOf(\InvalidArgumentException::class);
+   }
+
+   public function testGetTabNameForItem_notEnabledOrITIL() {
+      $this->clearImpactItems();
+      $impact = new \Impact();
+
+      $this->exception(function() use ($impact) {
+         $notEnabledOrITIL = new \ImpactCompound();
+         $impact->getTabNameForItem($notEnabledOrITIL);
+      })->isInstanceOf(\InvalidArgumentException::class);
+   }
+
+   public function testGetTabNameForItem_tabCountDisabled() {
+      $this->clearImpactItems();
+      $oldSession = $_SESSION['glpishow_count_on_tabs'];
+      $_SESSION['glpishow_count_on_tabs'] = false;
+
+      $impact = new \Impact();
+      $computer = new \Computer();
+
+      $this->string($impact->getTabNameForItem($computer))
+         ->isEqualTo("Impacts ");
+
+      $_SESSION['glpishow_count_on_tabs'] = $oldSession;
+   }
+
+   public function testGetTabNameForItem_enabledAsset() {
+      $this->clearImpactItems();
+      $oldSession = $_SESSION['glpishow_count_on_tabs'];
+      $_SESSION['glpishow_count_on_tabs'] = true;
+
+      $impact = new \Impact();
+      $impactRelationManager = new \ImpactRelation();
+
+      // Get computer
+      $computer = getItemByTypeName('Computer', '_test_pc02');
+
+      // Create an impact graph
+      $impactRelationManager->add([
+         'itemtype_source'   => "Computer",
+         'items_id_source'   => "1",
```

Comment:
You should use real id value. It may happen (on developper machine) that db contains altered bootstrap data. In this case, your test may fail.

To get id of a computer, you can use `getItemByTypeName('Computer', '_test_pc01', true);` (third param is used to retrieve only id).